src/com/palmergames/bukkit/towny/command/TownCommand.java

```diff
@@ -3848,9 +3833,6 @@ private static void sendTownMergeRequest(CommandSender sender, Town remainingTow
 		TownyMessaging.sendMsg(succumbingTown.getMayor(), Translatable.of("msg_town_merge_request_received", remainingTown.getName(), sender.getName(), remainingTown.getName()));
 
 		Confirmation.runOnAccept(() -> {
-			if (cost > 0 && TownyEconomyHandler.isActive() && !remainingTown.getAccount().canPayFromHoldings(cost))
-				TownyMessaging.sendErrorMsg(sender, Translatable.of("msg_town_merge_err_not_enough_money", (int) remainingTown.getAccount().getHoldingBalance(), (int) cost));
-
 			TownPreMergeEvent townPreMergeEvent = new TownPreMergeEvent(remainingTown, succumbingTown);
 			Bukkit.getPluginManager().callEvent(townPreMergeEvent);
 			if (townPreMergeEvent.isCancelled()) {

```

Comment:
If the pre merge event is cancelled here then the cost is never refunded back to the player.

Reply:
This has been sorted out, with the Confirmations returning to their previous ways.